source/includes/options-mongosqld.yaml

```diff
@@ -101,23 +101,57 @@ description: |
   ``admin`` and ``local`` databases.
 ---
 program: mongosqld
-name: sampleSource
+name: schemaSource
 directive: option
 args: <db-name>
 description: |
-  .. versionadded:: 2.3
+  .. versionchanged:: 2.11
+
+     Renamed ``--sampleSource`` to ``-schemaSource``     
 
-  Required when :option:`--sampleMode <mongosqld --sampleMode>` is set
-  to write.   Specifies a database to store schema information.
+  Required when :option:`--schemaMode <mongosqld --schemaMode>` is set
+  to ``auto``. Specifies a database to store schema information.
   
   .. note::
 
      If you do not specify either a schema file with the
      :option:`--schema` option or a database with the
-     ``--sampleSource`` option, :binary:`~bin.mongosqld` holds its
+     ``--schemaSource`` option, :binary:`~bin.mongosqld` holds its
      schema in memory.
 
   .. include:: /includes/sampling-ref-chart-link.rst
+---
+program: mongosqld
+name: schemaName
+directive: option
+args: <db-name>
+default: "``defaultSchema``"
+description: |
+  .. versionadded:: 2.11    
+
+  Optional. The name of the schema to load from or write to
+  the :option:`--schemaSource <mongosqld --schemaSource>` database
+  depending on the value of :option:`--schemaMode <mongosqld --schemaMode>`:
+
+  .. list-table::
+     :widths: 10 90
+     :header-rows: 1
+
+     * - :option:`--schemaMode <mongosqld --schemaMode>` Value
+       - ``--schemaName`` Behavior
+
+     * - ``custom``
+       - The name of the schema to load from the database specified
+         by the :option:`--schemaSource <mongosqld --schemaSource>`
+         option.
+
+     * - ``auto``
+       - The name of the schema to write to the :option:`--schemaSource
+         <mongosqld --schemaSource>`
```

Comment:
I think we need to flesh out what's happening here a little more. If I'm understanding this correctly, "--schemaName foo --schemaMode auto" will sample a mongodb database and store the schema information in a new collection called foo. Is that right? Either way we could be more explicit about it. The Sampling Mode Reference Chart is helpful, but it doesn't include the new --schemaName option.

Reply:
Being that these are so intertwined, I think we really need to defer to the chart. I did tighten this section up a bit.

Let me know if there are any other ways I can clarify.